app/Platform/Services/GameSuggestions/GameSuggestionEngine.php

```diff
@@ -0,0 +1,133 @@
+                [new Strategies\SimilarGameStrategy($this->sourceGame, attachContext: false), 50],
+                [new Strategies\SharedHubStrategy($this->sourceGame), 20],
+                [new Strategies\CommonPlayersStrategy($this->user, $this->sourceGame, attachContext: false), 20],
+                [new Strategies\SharedAuthorStrategy($this->sourceGame), 10],
+            ];
+        } else {
+            // Get the user's mastered and beaten games.
+            $masteredGames = Game::query()
+                ->whereHas('playerGames', function ($query) {
+                    $query->whereUserId($this->user->id)
+                        ->whereColumn('achievements_unlocked', 'achievements_total')
+                        ->where('achievements_total', '>', 0);
+                })
+                ->get();
+
+            $beatenGames = Game::query()
+                ->whereHas('playerGames', function ($query) {
+                    $query->whereUserId($this->user->id)
+                        ->whereNotNull('beaten_at');
+                })
+                ->get();
+
+            $this->strategies = [
+                [new Strategies\WantToPlayStrategy($this->user), 30],
+                [new Strategies\RevisedGameStrategy($this->user), 20],
+                [new Strategies\RandomGameStrategy(), 10],
+            ];
+
+            // Add strategies based on mastered games.
+            foreach ($masteredGames as $masteredGame) {
+                $weight = 50 / max(1, count($masteredGames));
+                $this->strategies[] = [new Strategies\SimilarGameStrategy($masteredGame), $weight];
+                $this->strategies[] = [new Strategies\SharedHubStrategy($masteredGame), $weight * 0.4];
+                $this->strategies[] = [new Strategies\SharedAuthorStrategy($masteredGame), $weight * 0.2];
+                $this->strategies[] = [new Strategies\CommonPlayersStrategy($this->user, $masteredGame), $weight * 0.4];
+            }
+
+            // Add strategies based on beaten games.
+            foreach ($beatenGames as $beatenGame) {
+                $weight = 25 / max(1, count($beatenGames));
+                $this->strategies[] = [new Strategies\SimilarGameStrategy($beatenGame), $weight];
+                $this->strategies[] = [new Strategies\SharedHubStrategy($beatenGame), $weight * 0.4];
+                $this->strategies[] = [new Strategies\SharedAuthorStrategy($beatenGame), $weight * 0.2];
+                $this->strategies[] = [new Strategies\CommonPlayersStrategy($this->user, $beatenGame), $weight * 0.4];
+            }
+        }
+    }
+
+    /**
+     * @return array<GameSuggestionData>
+     */
+    public function selectSuggestions(int $limit = 10): array
+    {
+        // We won't show the user games they already have 100% completion for.
+        $masteredGameIds = PlayerGame::whereUserId($this->user->id)
+            ->whereAllAchievementsUnlocked()
+            ->pluck('game_id')
+            ->toArray();
```

Comment:
I think I missed this in the original commit. I noticed when refreshing constantly in prod, I was never suggested a mastered game. This new logic should filter those out.